src/transformers/pipelines.py

```diff
@@ -2058,6 +2057,267 @@ def span_to_answer(self, text: str, start: int, end: int) -> Dict[str, Union[str
         }
 
 
+class TablequestionAnsweringArgumentHandler(ArgumentHandler):
+    """
+    Handles arguments for the TableQuestionAnsweringPipeline
+    """
+
+    def __call__(self, table=None, query=None, sequential=False, padding=True, truncation=True):
+        # Returns tqa_pipeline_inputs of shape:
+        # [
+        #   {"table": pd.DataFrame, "query": List[str]},
+        #   ...,
+        #   {"table": pd.DataFrame, "query" : List[str]}
+        # ]
+        import pandas as pd
+
+        if table is None:
+            raise ValueError("Table cannot be None.")
+        elif query is None:
+            if isinstance(table, dict) and table.get("query") is not None and table.get("table") is not None:
+                tqa_pipeline_inputs = [table]
+            elif isinstance(table, list) and len(table) > 0 and isinstance(table[0], dict):
+                if table[0].get("query") is not None and table[0].get("table") is not None:
+                    tqa_pipeline_inputs = table
+                else:
+                    raise ValueError(
+                        "If the first argument is a list, it should contain dictionaries containing "
+                        "`table` and `query` keys."
+                    )
+            else:
+                raise ValueError("Invalid input. Either query or table is None, or badly formed input in list.")
+        else:
+            tqa_pipeline_inputs = [{"table": table, "query": query}]
+
+        for tqa_pipeline_input in tqa_pipeline_inputs:
+            if not isinstance(tqa_pipeline_input["table"], pd.DataFrame):
+                if tqa_pipeline_input["table"] is None:
+                    raise ValueError("Table cannot be None.")
+
+                tqa_pipeline_input["table"] = pd.DataFrame(tqa_pipeline_input["table"])
+
+        return tqa_pipeline_inputs, sequential, padding, truncation
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class TableQuestionAnsweringPipeline(Pipeline):
+    """
+    Table Question Answering pipeline using a :obj:`ModelForQuestionAnswering`. As per the time of writing, the only
+    model that can be used is a model based off of the TAPAS architecture.
```

Comment:
```suggestion
    Table Question Answering pipeline using a :obj:`ModelForTableQuestionAnswering`.
```
Not fond of "at the tome of writing". The docs are supposed to be up to date :-p But we do need a special type of model for table QA I believe.